File: mo-common/mo-common-core/src/main/java/ink/ckx/mo/common/core/base/BaseEntity.java
package ink.ckx.mo.common.core.base;

import com.baomidou.mybatisplus.annotation.FieldFill;
import com.baomidou.mybatisplus.annotation.TableField;
import io.swagger.v3.oas.annotations.media.Schema;
import lombok.Data;

import java.io.Serial;
import java.io.Serializable;
import java.time.LocalDateTime;

@Schema(description = "实体基础类")
@Data
public class BaseEntity implements Serializable {

    @Serial
    private static final long serialVersionUID = 1L;

    @Schema(description = "是否删除")
    @TableField(fill = FieldFill.INSERT_UPDATE)
    private Integer deleted;

    @Schema(description = "更新时间")
    @TableField(fill = FieldFill.INSERT_UPDATE)
    private LocalDateTime updateTime;

    @Schema(description = "创建时间")
    @TableField(fill = FieldFill.INSERT)
    private LocalDateTime createTime;
}

File: mo-common/mo-common-core/src/main/kotlin/ink/ckx/mo/common/core/base/BaseEntity.kt
package ink.ckx.mo.common.core.base

import com.baomidou.mybatisplus.annotation.FieldFill
import com.baomidou.mybatisplus.annotation.TableField
import io.swagger.v3.oas.annotations.media.Schema
import java.io.Serializable
import java.time.LocalDateTime

@Schema(description = "实体基础类")
open class BaseEntity(

    @Schema(description = "是否删除")
    @TableField(fill = FieldFill.INSERT)
    var deleted: Int? = 0,

    @Schema(description = "更新时间")
    @TableField(fill = FieldFill.INSERT_UPDATE)
    var updateTime: LocalDateTime? = null,

    @Schema(description = "创建时间")
    @TableField(fill = FieldFill.INSERT)
    var createTime: LocalDateTime? = null,
) : Serializable